File: src/main/java/de/chaos/swlnmngr/route/routes/NewRoute.java
package de.chaos.swlnmngr.route.routes;

import de.chaos.swlnmngr.MainKt;
import de.chaos.swlnmngr.config.Config;

import java.io.File;
import java.io.IOException;
import java.nio.file.Files;

public class NewRoute implements Route {
    @Override
    public String getName() {
        return "new";
    }

    @Override
    public boolean route(String[] args) {
        if(args.length < 1) {
            MainKt.getLogger().error("Usage: swlnmngr [Name]");
            return false;
        }
        String name = args[0];
        File newFile = new File(Config.LIB_PATH, name);
        MainKt.getLogger().debug(newFile);
        if(newFile.exists()) {
            MainKt.getLogger().error("Directory {} already exists", name);
            return false;
        }
        newFile.mkdir();
        File defaultFile = new File(Config.LIB_PATH, Config.DEFAULTS);
        MainKt.getLogger().debug(defaultFile);
        try {
            for (File file : defaultFile.listFiles()) {
                MainKt.getLogger().debug(file);
                Files.createSymbolicLink(new File(newFile, file.getName()).toPath(), file.toPath());
            }
        } catch (IOException e) {
            MainKt.getLogger().error("Could not create SymLink", e);
            return false;
        }
        return true;
    }
}

File: src/main/java/de/chaos/swlnmngr/route/routes/NewRoute.kt
package de.chaos.swlnmngr.route.routes

import de.chaos.swlnmngr.config.config
import de.chaos.swlnmngr.logger
import org.apache.commons.io.FileUtils.listFiles
import org.apache.commons.io.filefilter.FileFilterUtils
import java.io.File
import java.nio.file.Files

class NewRoute :Route{
    override fun getName(): String = "new"

    override fun route(args: Array<out String>): Boolean {
        if (args.isEmpty()) {
            logger.error("Usage: swlnmngr new [Name]")
            return false
        }

        val name = args.component1()
        val newFile = File(config.libPath, name)
        logger.debug(newFile)
        if(newFile.exists()) {
            logger.error("Directory {} already exists", name)
            return false
        }

        newFile.mkdir()
        val defaultFile = File(config.libPath, config.default)
        logger.debug(defaultFile)
        for (file in listFiles(defaultFile, FileFilterUtils.trueFileFilter(), FileFilterUtils.trueFileFilter())) {
            logger.debug(file)
            Files.createSymbolicLink(File(newFile, file.name).toPath(), file.toPath())
        }

        return true
    }

}
